homework4/task41/MyArrayList.java

```diff
@@ -0,0 +1,66 @@
+package com.mateakademija.homework.homework4.task41;
+
+public class MyArrayList<E> implements SList <E> {
+
+    public static void main (String[] args) {
+      SList<String> str = new MyArrayList<>();
+            str.add("first");
+        str.add("second");
+        str.add("wwwwww");
+        str.add("tt");
+        str.remove(0);
+        System.out.println(str.get(2));
+    }
+
+
+    E[] values;
+    private int index = 0;
+
+    public MyArrayList () {
+        values = (E[]) new Object[0];
+    }
+
+    @Override
+    public int size () {
+        return values.length;
+    }
+
+    @Override
+    public E get (int index) {
+        return  values [index];
+    }
+
+    @Override
+    public void add (E e) {
+        try {
+            E[] temp = values;
+            values = (E[]) new Object[temp.length + 1];
```

Comment:
it is not a good practice to create a nea array everytime you add value
create array with default value and in add method just check if real size of array >= array.size